File: android/src/main/java/ro/tudorluca/gpstracks/android/Position.java
package ro.tudorluca.gpstracks.android;

import java.util.Date;

/**
 * Created by Tudor Luca on 12/12/14.
 */
public class Position {

    private long mUserId;
    private String mLatitude;
    private String mLongitude;
    private Date mDate;

    public Position() {
    }

    public Position(long userId, String latitude, String longitude, Date date) {
        mUserId = userId;
        mLatitude = latitude;
        mLongitude = longitude;
        mDate = date;
    }

    public long getUserId() {
        return mUserId;
    }

    public void setUserId(long userId) {
        mUserId = userId;
    }

    public String getLatitude() {
        return mLatitude;
    }

    public void setLatitude(String latitude) {
        mLatitude = latitude;
    }

    public String getLongitude() {
        return mLongitude;
    }

    public void setLongitude(String longitude) {
        mLongitude = longitude;
    }

    public Date getDate() {
        return mDate;
    }

    public void setDate(Date date) {
        mDate = date;
    }
}

File: webapi/src/main/kotlin/ro/tudorluca/gpstracks/webapi/core/Position.kt
package ro.tudorluca.gpstracks.webapi.core

import kotlin.properties.Delegates
import java.util.Date

/**
 * Created by Tudor Luca on 03/12/14.
 */
public class Position : BaseEntity() {

    class object {

        fun init(id: Long, dateCreated: Date, userId: Long, latitude: String, longitude: String, date: Date): Position {
            val position = Position()
            position.id = id;
            position.dateCreated = dateCreated
            position.userId = userId
            position.latitude = latitude
            position.longitude = longitude
            position.date = date

            return position
        }
    }

    public var userId: Long by Delegates.notNull()
    public var latitude: String by Delegates.notNull()
    public var longitude: String by Delegates.notNull()
    public var date: Date by Delegates.notNull()
}